index.html

```diff
@@ -72,100 +74,160 @@
         return (date.getUTCMinutes() > 0 ? date.getUTCMinutes() + "m " : "") + date.getUTCSeconds() + "s";
       }
 
-      function getImage(p) {
-        var name = p.name;
-        if (p.name == "Nidoran F") {
-          name = "Nidoran";
-        }
-        else if (name == "Nidoran M") {
-          name = "Nidorano";
-        }
-        return "http://icons.iconarchive.com/icons/hektakun/pokemon/72/" + pad(p.id, 3) + "-" + cap(name) + "-icon.png";
-      }
-
       function toggleLayer(checked, pkmn) {
         if (checked) {
-            hidden_layers.push(pkmn);
+            hidden_layers[pkmn] = 1;
         } else {
-            var index = hidden_layers.indexOf(pkmn);
-            if(index > -1){
-              hidden_layers.splice(index,1);
-            }
+            delete hidden_layers[pkmn];
```

Comment:
My fault. I did not see that you've changed hidden_layer from array to object.
delete is fine then.